extensions/cxf/runtime/src/main/java/io/quarkus/cxf/runtime/CXFQuarkusServlet.java

```diff
@@ -0,0 +1,75 @@
+package io.quarkus.cxf.runtime;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javax.servlet.ServletConfig;
+
+import org.apache.cxf.Bus;
+import org.apache.cxf.BusFactory;
+import org.apache.cxf.frontend.ServerFactoryBean;
+import org.apache.cxf.transport.servlet.CXFNonSpringServlet;
+import org.jboss.logging.Logger;
+
+import io.quarkus.arc.Arc;
+
+public class CXFQuarkusServlet extends CXFNonSpringServlet {
+
+    private static final Logger LOGGER = Logger.getLogger(CXFQuarkusServlet.class);
+
+    private static final List<WebServiceConfig> WEB_SERVICES = new ArrayList<>();
+
+    @Override
+    public void loadBus(ServletConfig servletConfig) {
+        super.loadBus(servletConfig);
+
+        // You could add the endpoint publish codes here
+        Bus bus = getBus();
+        BusFactory.setDefaultBus(bus);
+
+        // You can also use the simple frontend API to do this
+        ServerFactoryBean factory = new ServerFactoryBean();
+        factory.setBus(bus);
+
+        for (WebServiceConfig config : WEB_SERVICES) {
+            Object instanceService = Arc.container().instance(config.getClassName()).get();
+            if (instanceService != null) {
+                factory.setServiceBean(instanceService);
+                factory.setAddress(config.getPath());
+                factory.create();
+                LOGGER.info(config.toString() + " available.");
+            } else {
+                LOGGER.error("Cannot initialize " + config.toString());
+            }
+        }
```

Comment:
I don't understand this piece of code, looks like you use a single ServerFactoryBean for multiple webservice configuration ?

Reply:
Maybe we can rename factory to builder and use builder pattern ?

Reply:
Ah it's a builder and you reuse it, I understand now :)